### DISCUSSION TITLE
How long should migration be supported?

### DISCUSSION THREAD

**Discussion by WernerderChamp** [iota-community/Community-Governance#24] [Ideas]
In the #ideas-discussion chat, it was stated that unmigrated tokens (from Chrysalis) will be moved into the treasury as well.

As of now, only 31% of tokens are migrated (23.8% pre-launch). While I do expect a boost once ledger is implemented, there are probably a lot of users who just invested into IOTA, withdrew their tokens to be on the safe side and then just don't look into it for a long time.
Or they even forgot, that they've bought IOTA some time ago. I saw several users in the help chat asking why the light wallet does not work and how they should migrate. Trinity has been out for almost 3 years.

I think, just one year is not enough for a migration period, and we therefore have to think about how we are going to deal with late claimers. The legacy WOTS signature scheme, regardless if it is curl based (before mid 2017) or kerl based (mid 2017 until now) still allows users to verify their ownership, unless their address has been spent from twice already. Using a SC, those users could claim their legacy tokens with ease, so there is no need for central governance.

However, I do not know if all of the tokens in the treasury are from those curl->kerl migration, some of them seem to come all the way from the ICO. Also I read that the IF claimed tokens of addresses which were at risk due to too many double spends. Those tokens might be hard or even impossible to return to their rightful owners.

I think that, in general, returning tokens to their owners has priority over any other usecase. My proposal would therefore be to add reclaim-pools for tokens that are easy to reclaim. Every year, we take 10% of the balance out of the pools and move it too the DAO. This still provides us with a lot of tokens, and users have a chance of reclaiming their tokens. With every year, the chance of a token in the pool not getting claimed ever rises, therefore we can slowly but steadily drain.

Tokens from the Chrysalis migration should be left alone for at least 2 years, so everyone has a chance to migrate. I run a validator node and I'll be monitoring the situation closely.

For the tokens that cannot be easily migrated, I'm open for ideas and additional information from IF regarding the origin of these tokens.

**Comment by Phyloiota** [comment]
I really like your idea with the time based reducing of reclaim pools. 
I also think that we could find a solution for users that have been victims of protocol flaws (kerl/curl, moonpay) to be compensated.
From my understanding the vast amount of tokens in the treasury wallet is from the initial token sale (unclaimed from genesis) and therefore no one has ever had a seed of those tokens. Kerl / Curl seems to be a rather small fraction, but i would really first like to get proper information on this numbers.

**Reply by basvansambeek** [reply]
I have tokens take from me in 2017 that were vulnerable due to double spends. (Because Bitfinex required white listed addresses. Grmbl.)

Could it be that these are in custody with IOTA Foundation. I noticed that they were not moved for a long time.

**Reply by Phyloiota** [reply]
> I have tokens take from me in 2017 that were vulnerable due to double spends. (Because Bitfinex required white listed addresses. Grmbl.)
> 
> Could it be that these are in custody with IOTA Foundation. I noticed that they were not moved for a long time.

you can check that Bas: https://github.com/iota-community/IOTA-Discord/blob/main/help/iota-recovery-tips.md#reclaim
Or with that tool https://github.com/HBMY289/iotaZeroBalanceHelper
